parser.py

```diff
@@ -53,6 +54,22 @@ def tweet_json_to_markdown(tweet, username, archive_media_folder, output_media_f
         for url in tweet['entities']['urls']:
             if 'url' in url and 'expanded_url' in url:
                 body = body.replace(url['url'], url['expanded_url'])
+    # if the tweet is a reply, construct a header that links the names of the accounts being replied to the tweet being replied to
+    header = ''
+    if 'in_reply_to_status_id' in tweet:
+        # match and remove all occurences of '@username ' at the start of the body
+        replying_to = re.match(r'^(@[0-9A-Za-z_]* )*', body)[0]
+        body = body.removeprefix(replying_to)
+        if not replying_to:
+            # no '@username ' in the body: we're replying to self
+            replying_to = f'@{username}'
+        names = replying_to.split()
+        # some old tweets lack 'in_reply_to_screen_name': use it if present, otherwise fall back to names[0]
+        in_reply_to_screen_name = tweet['in_reply_to_screen_name'] if 'in_reply_to_screen_name' in tweet else names[0]
+        # create a list of names of the form '@name1, @name2 and @name3' - or just '@name1' if there is only one name
+        name_list = ', '.join(names[:-1]) + (f' and {names[-1]}' if len(names) > 1 else names[0])

```

Comment:
I've just realized that this change introduced English-language words where previously the content of the markdown was language-agnostic. I wonder if we should use `→` instead of `Replying to` and `,` instead of `and`?

Reply:
> I've just realized that this change introduced English-language words where previously the content of the markdown was language-agnostic. 

"Originally on Twitter" says ~hello~ goodbye :grin: 

> I wonder if we should use `→` instead of `Replying to` and `,` instead of `and`?

Maybe? I'd prefer not :slightly_smiling_face: 

Even commas are a language-specific construct. 

But maybe the better option would be to leave the space-separated "@name_1 @name_2 @name_n " at the start of each tweet, and hyperlink that to the original tweet being replied to?

Reply-to-self would need something different, because those tweet bodies don't start with your own username. I have a change in progress for reply-to-self threads, and was already thinking about how to handle linking between tweets in threads. If all goes well, I'll share a PR later today for discussion.

Reply:
All true. Thanks.